File: python-solutions/1482-how-many-numbers-are-smaller-than-the-current-number/solution.cpp
class Solution {
public:
    vector<int> smallerNumbersThanCurrent(vector<int>& nums) {

        std::vector<int> freq(101,0);

        for (int n : nums){
            freq[n]++;
        }

        for (int i = 1; i < 101; i++){
            freq[i] = freq[i-1]+freq[i];
        }

        std::vector<int> output;
        for (int num: nums){
            if (num == 0){
                output.push_back(0);
            }
            else{
                output.push_back(freq[num-1]);
            }

        }
        return output;
    }
};

File: python-solutions/1482-how-many-numbers-are-smaller-than-the-current-number/solution.py
class Solution:
    def smallerNumbersThanCurrent(self, nums: List[int]) -> List[int]:
        freq = [0]*101
        for num in nums:
            freq[num] += 1

        for i in range(1,101):
            freq[i] = freq[i-1]+freq[i]

        output = []
        for j in nums:
            if j == 0:
                output.append(0)
            else:    
                output.append(freq[j-1])
        
        return output
            
